ambry-clustermap/src/main/java/com/github/ambry/clustermap/PropertyStoreToDataNodeConfigAdapter.java

```diff
@@ -142,14 +162,19 @@ public void onDataCreate(String path) {
 
     @Override
     public void onDataDelete(String path) {
-      // TODO handle node deletions dynamically. Doing so requires further work in the ClusterChangeHandler impl
-      LOGGER.info("DataNodeConfig path {} deleted. This requires a restart to handle", path);
+      // Handle node deletions dynamically.
+      LOGGER.info("DataNodeConfig path {} deleted", path);
+      onPathDelete(path);
```

Comment:
single usage one-liner fn can be replaced with the contents of the fn

Reply:
In this case, I felt it is better to not inline it since it matches the pattern like you said.